### DISCUSSION TITLE
some questiones about prisma nested transaction

### DISCUSSION THREAD

**Discussion by BarrySJL** [prisma/prisma#10619] [Q&A] [answered]
I am very new to prisma. when I use transaction. I met some problems. look at the test code
![Screen Shot 2021-12-08 at 3 05 18 PM](https://user-images.githubusercontent.com/26754558/145163753-cffd7126-184f-4583-98e8-c0a8b9408245.png)

the record which id=3 has been existed in the database, so the code will be executed throw an error. no data in database will be changed.

but when I change the order test1() and test2() function.
![Screen Shot 2021-12-08 at 3 23 41 PM](https://user-images.githubusercontent.com/26754558/145165977-a0211cf1-9f7e-4bed-a777-9cdcbb5abdd7.png)

the record which id=3 has been updated in database. although the test1 throw an error.
so this is the question, why the transaction is not rollback. I know I use nested transaction. 

another question.
![Screen Shot 2021-12-08 at 3 31 14 PM](https://user-images.githubusercontent.com/26754558/145166910-b8b606ad-c7e4-43e7-8d68-a205568f3e31.png)

I use try catch around the create(id=3). just in one transaction . the transaction also not rollback.
if remove try catch block the transaction works well.
It is not same with JAVA. in java no matter use try catch or not . the transaction will be rollback.
so I am very confused now. 

does the transaction have hooks like "afterCommmit" "afterRollback"?

**Comment by austincrim** [comment, marked as answer]
Hey @BarrySJL 👋 , sorry for the late response.

I was not able to recreate this issue locally. Could you provide me with a minimal reproduction repo so that I can continue to investigate?

> does the transaction have hooks like "afterCommmit" "afterRollback"?

Prisma interactive transactions do not have these hooks but there is an [open feature request](https://github.com/prisma/prisma/issues/9083) for them. Feel free to add your +1 to help our engineers prioritize!

**Reply by sunnywhy** [reply]
You can try testing with Postgres, I can quickly reproduce it with Postgres.

**Comment by coler-j** [comment]
I don't think there is support for nested transactions is there?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
